src/main/scala/com/fulcrumgenomics/vcf/filtration/ReadEndSomaticVariantFilter.scala

```diff
@@ -123,27 +124,117 @@ class EndRepairArtifactLikelihoodFilter(val distance: Int = 2,
 
     Map(Info.id -> pMutation)
   }
+}
+
+/** The companion object for [[ReadEndSomaticVariantFilter]]. */
+object ReadEndSomaticVariantFilter {
+
+  /** Returns true if <gt> is a heterozygous SNV with one reference allele, else false */
+  private[filtration] def isPointMismatch(gt: Genotype) : Boolean = {
+    val ref = gt.alleles.ref
+    if (ref.length != 1) false else { // No interest in deletions/MNVs, so only consider if ref length is 1
+      gt
+        .alleles
+        .alts
+        .collect { // Collect all the alt alleles
+          case alt: SimpleAllele // Only interested in simple alleles, no symbolics or other strange stuff
+            if (alt.length == 1 & alt.bases != ref.bases) => alt // No interest in insertions, so only retain alts with length 1
+          }.nonEmpty // If anything remains, at least one of our alleles is a point mismatch
+    }
+  }
 
   /** Calculates a pair of priors:
-    *   The first is the prior that the genotype is the result of a true somatic mutations
-    *   The second is the prior that the genotype is the result of the end-repair artifact process
+    *   <priorMutation> is the prior that the genotype is the result of a true somatic mutation.
+    *   <priorArtifact> is the prior that the genotype is an artifact resulting from the mechanism specific to the filter
+    *
     *
     * Default implementation gives a mutation prior of {{{(maf * 2)^2}}} capped at 0.9999 and an
     * artifact prior of 1 - mutation prior.  This has the effect of creating priors that are heavily
     * skewed towards the artifact at very low MAFs, and towards mutations as the MAF increases towards 0.5.
     */
-  private[filtration] def priors(pileup: Pileup[PileupEntry], maf: Double) : (Double, Double) = {
+  private[filtration] def priors(pileup: Pileup[PileupEntry], maf: Double, logger : Option[Logger] = None) : (Double, Double) = {
```

Comment:
I don't love passing `logger` objects around in function calls.  Any reason not to just have the `ReadEndSomaticVariantFilter` object extend `LazyLogging` and use it's own logger?

Reply:
resolved